File: ktx/src/main/java/ru/surfstudio/android/ktx/util/KeyboardUtil.java
package ru.surfstudio.android.ktx.util;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

/**
 * Утилита для работы с экранной клавиатурой
 */
public class KeyboardUtil {

    /**
     * Показать экранную клавиатуру. <p>
     * Клавиатура открывается с нужным для указанного {@link EditText} {@link android.text.InputType}.
     *
     * @param editText {@link EditText}, для которого открывается клавиатура
     */

    public static void showKeyboard(EditText editText) {
        InputMethodManager imm = (InputMethodManager) editText.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
        editText.postDelayed(() -> imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT), 50);
    }

    /**
     * Скрыть экранную клавиатуру
     */
    public static void hideSoftKeyboard(Activity activity) {
        InputMethodManager inputMethodManager = (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
        if (activity.getCurrentFocus() != null) {
            inputMethodManager.hideSoftInputFromWindow(activity.getCurrentFocus().getWindowToken(), 0);
        }
    }

    /**
     * Скрыть экранную клавиатуру с вьюшки
     */
    public static void hideSoftKeyboard(View v) {
        InputMethodManager imm = (InputMethodManager) v.getContext().getSystemService(Activity.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
    }

    private KeyboardUtil() {
        throw new IllegalStateException(KeyboardUtil.class + " не может иметь инстансов");
    }
}

File: util-ktx/src/main/java/ru/surfstudio/android/util/KeyboardUtil.kt
package ru.surfstudio.android.util

import android.app.Activity
import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

/**
 * Утилита для работы с экранной клавиатурой
 */
object KeyboardUtil {

    /**
     * Показать экранную клавиатуру.
     *
     *
     * Клавиатура открывается с нужным для указанного [EditText] [android.text.InputType].
     *
     * @param editText [EditText], для которого открывается клавиатура
     */

    fun showKeyboard(editText: EditText) {
        val imm = editText.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        editText.postDelayed({ imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT) }, 50)
    }

    /**
     * Скрыть экранную клавиатуру
     */
    fun hideSoftKeyboard(activity: Activity) {
        val view = activity.currentFocus
        if (view != null) {
            val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }

    /**
     * Скрыть экранную клавиатуру с вьюшки
     */
    fun hideSoftKeyboard(v: View) {
        val imm = v.context.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(v.windowToken, 0)
    }

}